### PR TITLE
Open widget framework integration

### PR DESCRIPTION
**Pull request by jklingen92**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots 
  - [ ] Mobile width screenshots 
  - [ ] tag @ferdi or @pdpinch for review  
- [ ] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
#1441 

#### What's this PR do?
Initial integration of the open-widget-framework

#### How should this be manually tested?
No testing yet

#### Where should the reviewer start?
See if you can locally view, edit, add, and delete widgets

#### Any background context you want to provide?
There's still a lot to change and adjust. If the input is invalid (URLs especially) it won't add the widget

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
(Optional)

### CODE REVIEW

File: open_discussions/templates/base.html [hunk 1]
```diff
@@ -13,6 +13,8 @@
     {% if js_settings_json %}
     <script type="text/javascript">
     var SETTINGS = {{ js_settings_json|safe }};
+    var csrfToken = '{{ csrf_token }}';
```
Comment (pdpinch) [inline open_discussions/templates/base.html]: what's this for / is it necessary?

File: static/js/containers/App.js [hunk 2]
```diff
@@ -292,6 +294,9 @@ class App extends React.Component<AppProps> {
             path={`${match.url}privacy-statement`}
             component={PrivacyPolicyPage}
           />
+          <Route path={match.url} render={() => {
+            return (<WidgetList widgetListId={SETTINGS.widgetListId} />)
+          }}/>
```
Comment (pdpinch) [inline static/js/containers/App.js]: by putting this here, the widgets will appear on all pages, right?

File: static/js/components/WidgetWrappers.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    render() {
+      return (
+        <div>
+          {this.props.renderList(this.props.widgetInstances)}
+        </div>
+      )
+    }
+}
+
+class mitodlWidgetWrapper extends Component {
```
Comment (aliceriot) [inline static/js/components/WidgetWrappers.js]: are the components written in this file used anywhere?

File: static/js/containers/App.js (hunk 2 above)
Comment (jklingen92) [reply static/js/containers/App.js]: Correct.

File: static/js/components/WidgetWrappers.js (hunk 3 above)
Comment (jklingen92) [reply static/js/components/WidgetWrappers.js]: Those are stubs but haven't been fully implemented yet. They also will probably be nicer if / when we implement redux in the package and then we won't need to pass through so many widget props.

File: open_discussions/templates/base.html (hunk 1 above)
Comment (jklingen92) [reply open_discussions/templates/base.html]: That is for making requests to the django backend. It's a django generated token that the backend reads and verifies that it's not a cross-site request forgery. If your project has a different way of handling this, we can change it.

**Comment by pdpinch** [comment]
@jklingen92 this is a really helpful start for us. @rhysyngsun is going to take this PR from here to do some of the more idiosyncratic stuff.

**Comment by rhysyngsun** [comment]
Replaced by a newer PR #1534

**Pull request closed without merging**